### PR TITLE
CAMEL-11152 - set credentials for ssh connection through headers

### PR DESCRIPTION
**Pull request by onders86**
adoc updated and ut provided.

### CODE REVIEW

**Comment by oscerd** [comment]
@davsclaus Looks good to me.

File: components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SshConstants.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.ssh;
+
+public interface SshConstants {
+
+    String USERNAME_HEADER = "CamelSshUsernameHeader";
```
Comment (davsclaus) [inline components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SshConstants.java]: We usually do not use xxxHeader in the key constant

Comment (onders86) [inline components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SshConstants.java]: do we need to do it like 
for example;

String USERNAME_HEADER = "SshUsernameHeader";

Comment (davsclaus) [inline components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SshConstants.java]: More like

     USERNAME_HEADER = "SshUsername";

Comment (onders86) [inline components/camel-ssh/src/main/java/org/apache/camel/component/ssh/SshConstants.java]: ok. fixed the naming.

File: components/camel-ssh/src/main/docs/ssh-component.adoc [hunk 2]
```diff
@@ -131,7 +131,11 @@ and use that for authentication.
 `keyPairProvider` has been set, and if so, it will use that to for
 certificate based authentication.
 3.  If neither `certResource` nor `keyPairProvider` are set, it will use
-the `username` and `password` options for authentication.
+the `username` and `password` options for authentication. Even though the `username` 
+and `password` are provided in the endpoint configuration and headers set with 
+`SshConstants.USERNAME_HEADER` (`CamelSshUsernameHeader`) and
```
Comment (davsclaus) [inline components/camel-ssh/src/main/docs/ssh-component.adoc]: Update this doc as well

**Comment by davsclaus** [comment]
Thanks.

@oscerd looks good

**Comment by oscerd** [comment]
Thanks @onders86 the PR has been merged. Can you close this?

**Comment by onders86** [comment]
thanks. closing

**Pull request closed without merging**